glue/core/subset.py

```diff
@@ -1079,6 +1086,12 @@ def __init__(self, state1, state2=None):
     def copy(self):
         return type(self)(self.state1, self.state2)
 
+    def move_to(self, *args):
+        """Move any underlying ROI to the new given center."""
+        self.state1.move_to(*args)
+        if self.state2:
+            self.state2.move_to(*args)
```

Comment:
Thinking of your previous issue and comment on offset (non-concentric) AnnulusROIs, I am wondering if such `CompositeSubsetState`s could be constructed here. In that case the inner ROI would probably have to be moved as
```suggestion
           self.state2.move_to(self.state2.center()[0] - self.state1.center()[0] + x, self.state2.center()[1] - self.state1.center()[1] + y, *args)
```
...

Reply:
I don't understand your suggestion. I had tested this downstream and the annulus moved fine with my patch as-is. I am relying on the built-in state recursion.

Reply:
Oh... I just saw "non-concentric". Is that a thing? I have not seen such things implemented in both `regions` and `photutils`.

Reply:
Yes, not saying that this is presently implemented anywhere, but if it could be done in principle, better to allow for such a case.

Reply:
But the subset state currently does not have a `.center()` and I am not 100% all ROIs have consistent `.center()` methods. This is not something I want to deal with in this patch. I'd rather we open up a follow up issue for non-concentric support.

Reply:
Yes, `CircularROI` is lacking it, not sure of any others; but if it is not available for `self.state2` generally, this can not be added here (I guess that means `CompositeSubsetState`s with different centres cannot be constructed ATM).

Reply:
I mean, you could do composite subset state with whatever you want, and you correctly pointed out that `move_to` here won't work correctly for most of those except when they share the same center. I was hoping this won't be a problem for Jdaviz but now that I think about it... sigh, someone is going to draw something weird over there and encounter exactly this. 😭 

So, why don't you and Tom hash out what is the plan with https://github.com/glue-viz/glue/pull/2391#issuecomment-1524084015 and then we can try to address that here? Thanks!